### ISSUE TITLE
Whitespace issue with inline math in lists

### ISSUE THREAD

**Issue by prabhuramachandran** [alvinwan/TexSoup#20]
Noticed this little bug:
```
from TexSoup import TexSoup
s = TexSoup(r'''
\begin{itemize}
\item hello $\alpha$
\end{itemize}
''')
```
If you print this out, you'll see some important whitespace eaten up::
```
print(s)

\begin{itemize}
\item hello$\alpha$
\end{itemize}
```
The space after `hello` is gone.

**Comment by alvinwan** [comment]
Thanks for reporting! I refactored parts of the parser 42f7383 in addressing #21 and looks like this is fixed now

**Comment by prabhuramachandran** [comment]
Great, thanks!  I'll test later and let you know, will you be pushing a release to pypi sometime?

**Comment by alvinwan** [comment]
No problem. Yeah, in a sec here - right after I fix #22, I'll push v0.1.1
